execution/executor-benchmark/src/pipeline.rs

```diff
@@ -1,39 +1,44 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{
-    benchmark_transaction::BenchmarkTransaction, TransactionCommitter, TransactionExecutor,
-};
+use crate::{TransactionCommitter, TransactionExecutor};
 use aptos_executor::block_executor::{BlockExecutor, TransactionBlockExecutor};
 use aptos_executor_types::BlockExecutorTrait;
 use aptos_logger::info;
-use aptos_types::transaction::Version;
+use aptos_types::transaction::{Transaction, Version};
 use std::{
     marker::PhantomData,
-    sync::{mpsc, Arc},
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        mpsc, Arc,
+    },
     thread::JoinHandle,
+    time::Duration,
 };
 
 pub struct Pipeline<V> {
     join_handles: Vec<JoinHandle<()>>,
     phantom: PhantomData<V>,
+    start_commit: Arc<AtomicBool>,
 }
 
 impl<V> Pipeline<V>
 where
-    V: TransactionBlockExecutor<BenchmarkTransaction> + 'static,
+    V: TransactionBlockExecutor<Transaction> + 'static,
 {
     pub fn new(
-        executor: BlockExecutor<V, BenchmarkTransaction>,
+        executor: BlockExecutor<V, Transaction>,
         version: Version,
-    ) -> (Self, mpsc::SyncSender<Vec<BenchmarkTransaction>>) {
+        split_stages: bool,
+    ) -> (Self, mpsc::SyncSender<Vec<Transaction>>) {
         let parent_block_id = executor.committed_block_id();
         let executor_1 = Arc::new(executor);
         let executor_2 = executor_1.clone();
 
         let (block_sender, block_receiver) =
-            mpsc::sync_channel::<Vec<BenchmarkTransaction>>(50 /* bound */);
-        let (commit_sender, commit_receiver) = mpsc::sync_channel(3 /* bound */);
+            mpsc::sync_channel::<Vec<Transaction>>(50 /* bound */);
+        let (commit_sender, commit_receiver) =
+            mpsc::sync_channel(if split_stages { 10000 } else { 3 } /* bound */);
```

Comment:
Hmm.. this will cause memory issues if we have too many outstanding commits right?